pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,494 @@
+
+			//inserting reward info into the storage
+			<InitializeRewards<T>>::insert(reward_id, reward_info);
+
+			Self::deposit_event(Event::RewardCycleCreated { start_block, end_block, reward_id });
+
+			Ok(())
+		}
+
+		///The extrinsic will transfer and lock users rewards into users account
+		/// # Parameters,
+		/// * `origin`: The users address which has been mapped to reward id
+		/// * `id`: Reward id
+		#[pallet::weight(10_000)]
+		pub fn initialize_claim_rewards(origin: OriginFor<T>, reward_id: u32) -> DispatchResult {
+			let user: T::AccountId = ensure_signed(origin)?;
+			//check if given id valid or not
+			ensure!(
+				<InitializeRewards<T>>::contains_key(reward_id),
+				Error::<T>::RewardIdNotRegister
+			);
+
+			// check if rewards can be unlocked at current block
+			if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+				ensure!(
+					reward_info.start_block.saturated_into::<u128>() <=
+						<frame_system::Pallet<T>>::block_number().saturated_into::<u128>(),
+					Error::<T>::RewardsCannotBeUnlockYet
+				);
+			} else {
+				//sanity check
+				return Err(Error::<T>::RewardIdNotRegister.into())
+			}
+
+			//check if user has already initialize the reward
+			ensure!(
+				!<Distributor<T>>::contains_key(reward_id, &user),
+				Error::<T>::RewardsAlreadyInitialized
+			);
+
+			let account_in_vec: Vec<u8> = T::AccountId::encode(&user);
+
+			//get info of user from pre defined hash map
+			if let Some((total_rewards_in_pdex, initial_rewards_claimable, factor)) =
+				crowdloan_rewardees::HASHMAP.get(&account_in_vec)
+			{
+				//get reward info
+				if let Some(reward_info) = <InitializeRewards<T>>::take(reward_id) {
+					if *total_rewards_in_pdex > MIN_REWARDS_CLAIMABLE_AMOUNT {
+						//initialize reward info struct
+						let reward_info = RewardInfoForAccount {
+							total_reward_amount: (*total_rewards_in_pdex).saturated_into(),
+							claim_amount: 0_u128.saturated_into(),
+							is_initial_rewards_claimed: false,
+							is_initialized: false,
+							lock_id: REWARDS_LOCK_ID,
+							last_block_rewards_claim: reward_info.start_block,
+							initial_rewards_claimable: (*initial_rewards_claimable)
+								.saturated_into(),
+							factor: (*factor).saturated_into(),
+						};
+						//insert reward info into storage
+						<Distributor<T>>::insert(reward_id, user.clone(), reward_info);
+					}
+				} else {
+					//sanity check
+					return Err(Error::<T>::RewardIdNotRegister.into())
+				}
+			} else {
+				return Err(Error::<T>::UserNotEligible.into())
+			}
+
+			//check if user is added in reward list
+			ensure!(<Distributor<T>>::contains_key(reward_id, &user), Error::<T>::UserNotEligible);
+
+			<Distributor<T>>::mutate(reward_id, user.clone(), |user_reward_info| {
+				if let Some(user_reward_info) = user_reward_info {
+					//check if user already unlocked the rewards
+					ensure!(!user_reward_info.is_initialized, Error::<T>::RewardsAlreadyUnlocked);
```

Comment:
I see regression in tests comparing to the previous PR. [This](https://github.com/Polkadex-Substrate/Polkadex/blob/c4cf4a4ce9175bd3b76c0c658b02a789f6da237b/pallets/rewards/src/tests.rs#L778-L782) case is not covered here. Please add such assertion back.

Reply:
Added the test case, which helped me to find a bug, thanks.